src/Shared/AssemblyFolders/AssemblyFoldersEx.cs

```diff
@@ -258,6 +262,7 @@ OpenBaseKey openBaseKey
 
                     if (null != directoryName)
                     {
+                        _uniqueDirectoryPaths.Add(directoryName);
```

Comment:
Could be initialized lazily when first asked for but we seem to always construct `AssemblyFoldersExCache` after initializing `AssemblyFoldersEx` so the only scenario where it would make a difference is the (likely not heavily used) `GetAssemblyFoldersExInfo` public API. Plus the cost is negligible. Looks fine!